### PR TITLE
Simplify imports (#39)

### LINKED ISSUE #39: Simpler import structure
**Is your feature request related to a problem? Please describe.**
It's tedious to import all the stuff you might need from `apiron`:

```python
from apiron.service.base import Service
from apiron.endpoint import Endpoint, JsonEndpoint
from apiron.client import ServiceCaller
```

**Describe the solution you'd like**
Given the fairly small footprint of the features of this package, the common needs should be available to import from the top-level of the package:

```python
from apiron import Endpoint, JsonEndpoint, Service, ServiceCaller
```

**Additional context**
Use `__all__` and `__init__.py` to make things available without moving them. Things to include are probably:

```
Endpoint
JsonEndpoint
StreamingEndpoint
StubEndpoint

Service
DiscoverableService

APIException
NoHostsAvailableException
UnfulfilledParameterException

ServiceCaller
Timeout
```

### PR DESCRIPTION
**Pull request by daneah**
**This change is a:** (check at least one)
- [ ] Bugfix
- [ ] Feature addition
- [x] Code style update
- [ ] Refactor

**Is this a breaking change?** (check one)
- [ ] Yes
- [x] No

**Is the:**
- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite passing?
- [x] Code coverage maximal?

**What does this change address?**
Resolves #39

**How does this change work?**
Using a bit of `__all__` and `import *` magic ✨, make all the commonly-used stuff importable from the top level.

**Additional context**
Code within the implementation should *not* simplify its own imports, as this leads to circular references. This simplification is for tests and users only!

The tests continued to pass prior to simplifying their imports, so that confidence in the backward-compatibility allowed me to update them as well.

### CODE REVIEW

File: apiron/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+from apiron.endpoint import *
```
Comment (momo-sa) [inline apiron/__init__.py]: Why `import *` instead of documenting the imports explicitly as below for `client` and `exceptions`?

Comment (daneah) [reply apiron/__init__.py]: `client` and `exceptions` are modules, whereas `endpoint` and `service` are packages that define an `__all__`. The whole point of `__all__` is to enable this, really, so I figured I'd use it. I'd prefer to direct people to the actual documentation for understanding the full breadth of available features. Thoughts?

Comment (momo-sa) [reply apiron/__init__.py]: The whole point of `__all__` is that the linter won't freak out about unused imports in my init files. 😸 

```python
from foo import bar
__all__ = ['bar']
```

Jokes aside, avoiding `*` is largely a preference on my part---no strong reasons here. I often look at inits to see what's supposed to be exposed by a module or package. It's nice when it's all in one place. Plus, someone might want to expose different things at different layers, so it makes sense that sometimes `*` won't do, while being explicit always works. That, and cruel linters finally beat a revulsion of `import *` into me.

Anyway, @daneah your rationale makes sense. I was just curious. 😄

Comment (daneah) [reply apiron/__init__.py]: That's fair, and at some point in the future if it's clear we need to deliberately layer what we expose I'm all about designing that scheme. I don't prefer to use `import *` in higher-level code, but when it comes to lower-level libraries I think it serves a valid purpose.

**Pull request merged**